### PR TITLE
Add Shareable Link Modal to Download page

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

#9 

## Purpose/Implementation Notes

- Added a generic component for dialogs
- Added Shareable Link Modal to Download page 

We will still need to connect Redux to show the correct url.

## Types of changes

* [ ] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots

download

### CODE REVIEW

File: src/components/InputCopy/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      <div className="input-copy">
+        {this.state.alert && (
+          <div className="input-copy__alert">
+            <i className="ion-checkmark-circled" /> Copied to clipboard.
+          </div>
+        )}
+        <div>
+          <Button
+            text="Copy"
+            className="input-copy__btn"
```
Comment (ramenhog) [inline src/components/InputCopy/index.js]: From the screenshot it looks like a standard primary button, so you shouldn't have to pass in a custom `className`?

File: src/components/Input/Input.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+
+.input {
+    font-size: 16px;
+    line-height: 24px;
+    padding: 4px 8px;
+    border: 1px solid #9B9B9B;
```
Comment (ramenhog) [inline src/components/Input/Input.scss]: Let's replace all the hexcodes with sass variables in the `common/variables` file for consistency

File: src/components/Input/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import './Input.scss';
+
+type Props = {
```
Comment (ramenhog) [inline src/components/Input/index.js]: Nice! Type checking 🎉

File: src/components/Button/index.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
 import './Button.scss';
 
-const Button = ({ text, buttonStyle, onClick }) => {
+const Button = ({ text, buttonStyle, onClick, className }) => {
   const buttonStyleClass = buttonStyle ? `button--${buttonStyle}` : '';
   return (
-    <button className={`button ${buttonStyleClass}`} onClick={onClick}>
+    <button
+      className={`button ${buttonStyleClass} ${className}`}
```
Comment (ramenhog) [inline src/components/Button/index.js]: We should probably check to make sure className isn't `null` or `undefined`

File: src/components/InputCopy/index.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+export default class InputCopy extends React.Component {
+  state = {
+    alert: false
+  };
+
+  render() {
+    return (
+      <div className="input-copy">
+        {this.state.alert && (
```
Comment (ramenhog) [inline src/components/InputCopy/index.js]: 👍

File: src/components/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          className={`modal ${
+            this.props.center ? 'modal--center' : 'modal--top'
+          }`}
+          // Check that clicks were originated exactly on the `.modal` element, aka outside of the
+          // modal dialog. Clicks inside of the modal dialog will trigger this handler too.
+          onClick={e => e.target === e.currentTarget && this._closeModal()}
+        >
+          <div className={`modal__dialog ${this.props.className}`}>
+            {this.props.children}
+            <a
```
Comment (ramenhog) [inline src/components/Modal/index.js]: For accessibility purposes, let's actually change this to a button.

File: src/components/InputCopy/index.js (hunk 1 above)
Comment (arielsvn) [reply src/components/InputCopy/index.js]: hmm that class is used to position the button to the right. Do you think it's better to add an additional element containing the button and apply the class to that one instead?

Comment (ramenhog) [reply src/components/InputCopy/index.js]: hmm what about just doing `display: flex` on the parent?

Comment (arielsvn) [reply src/components/InputCopy/index.js]: haha great idea! I have to start using `flex` more

**Comment by arielsvn** [comment]
@ramenhog I addressed all the comments, let me know if you see something else

File: src/components/Modal/ModalManager.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    modalOpen: false
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        {this.props.component(() => this.setState({ modalOpen: true }))}
+        <Modal
+          {...this.props.modalProps}
+          key="2"
```
Comment (ramenhog) [inline src/components/Modal/ModalManager.js]: are we actually using this `key` anywhere?

File: src/components/Modal/index.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    this.el = document.createElement('div');
+  }
+
+  componentDidMount() {
+    modalRoot.appendChild(this.el);
+    this._addBodyClass();
+  }
+
+  componentDidUpdate() {
+    this._addBodyClass();
```
Comment (ramenhog) [inline src/components/Modal/index.js]: I'm wondering if we need a conditional check here to update the body class only when `isOpen` changes?

File: src/components/Modal/ModalManager.js (hunk 7 above)
Comment (arielsvn) [reply src/components/Modal/ModalManager.js]: No, we're not using it anymore, I forgot to remove it after refactoring that. Let me do it.

File: src/components/Modal/index.js (hunk 8 above)
Comment (arielsvn) [reply src/components/Modal/index.js]: good idea, that way we can avoid adding/removing the classes from the body unnecessarily.

**Comment by arielsvn** [comment]
yes me too, I like what we did there :)

**Pull request merged**